ethers-providers/src/transports/quorum.rs

```diff
@@ -437,7 +437,12 @@ where
         method: &str,
         params: T,
     ) -> Result<R, Self::Error> {
-        let mut params = serde_json::to_value(params)?;
+        let mut params = if std::mem::size_of::<T>() == 0 {
+            // we don't want `()` to become `"null"`.
+            serde_json::json!([])
+        } else {
+            serde_json::to_value(params)?
+        };
```

Comment:
I don't think this is the right, fix.

as you've pointed out the params field is skipped by `Request` if it's a zst, which is not the case for `serde_json::json!([])`.

we had the same issue with the RetryProvider that we fixed like this:

https://github.com/gakonst/ethers-rs/blob/92b51b6d9fbec759857a5af01437540bee5edde2/ethers-providers/src/transports/retry.rs#L143-L157

I suggest we extract this solution and make it applicable for the quorum provider as well.

Reply:
Made the changes, hopefully this is what you were thinking. Was debating between pulling it out fully and just kinda duplicating as I did and this felt okay given that in most places we can hide the details. If you want it fully abstracted please let me know where you think it should go.